core/server/utils/packages/read-packages.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Dependencies
+ */
+
+var parsePackageJson = require('./parse-package-json'),
+    Promise = require('bluebird'),
+    _ = require('lodash'),
+    join = require('path').join,
+    fs = require('fs'),
+
+    notAPackageRegex = /^\.|_messages|README.md|node_modules|bower_components/i,
+    packageJSONPath = 'package.json',
+
+    statFile = Promise.promisify(fs.stat),
+    readDir = Promise.promisify(fs.readdir),
+
+    readPackage,
+    readPackages,
+    processPackage;
+
+/**
+ * Recursively read directory and find the packages in it
+ */
+processPackage = function processPackage(absolutePath, packageName) {
+    var pkg = {
+        name: packageName,
+        path: absolutePath
+    };
+    return parsePackageJson(join(absolutePath, packageJSONPath))
+        .then(function gotPackageJSON(packageJSON) {
+            pkg['package.json'] = packageJSON;
+            return pkg;
+        })
+        .catch(function noPackageJSON() {
+            // ignore invalid package.json for now,
+            // because Ghost does not rely/use them at the moment
+            // in the future, this .catch() will need to be removed,
+            // so that error is thrown on invalid json syntax
+            pkg['package.json'] = null;
+            return pkg;
+        });
+};
+
+readPackage = function readPackage(packagePath, packageName) {
+    var absolutePath = join(packagePath, packageName);
+    return statFile(absolutePath)
+        .then(function (stat) {
+            if (!stat.isDirectory()) {
+                return {};
+            }
+
+            return processPackage(absolutePath, packageName)
+                .then(function gotPackage(pkg) {
+                    var res = {};
+                    res[packageName] = pkg;
+                    return res;
+                });
+        })
+        .catch(function () {
+            // @TODO: maybe log the error here?
```

Comment:
I would not return `{}` and instead return the error to the caller. And the caller can handle/decide.

e.g. if we read the active theme on bootstrap and it's not available - Ghost would log the error and show the user.